components/datetime/src/fieldsets.rs

```diff
@@ -1564,6 +1571,41 @@ impl_zone_marker!(
     input_tzid = yes,
 );
 
+impl_zone_marker!(
+    /// A time zone ID is required to format with this style.
+    /// For example, a raw [`UtcOffset`] cannot be used here.
+    ///
+    /// ```compile_fail,E0277
+    /// use icu::calendar::{DateTime, Iso};
+    /// use icu::datetime::FixedCalendarDateTimeFormatter;
+    /// use icu::datetime::fieldsets::X;
+    /// use icu::timezone::UtcOffset;
+    /// use tinystr::tinystr;
+    /// use icu::locale::locale;
+    /// use writeable::assert_writeable_eq;
+    ///
+    /// let utc_offset = UtcOffset::try_from_str("-06").unwrap();
+    ///
+    /// let formatter = FixedCalendarDateTimeFormatter::try_new(
+    ///     locale!("en-US").into(),
+    ///     X::new(),
+    /// )
+    /// .unwrap();
+    ///
+    /// // error[E0277]: the trait bound `UtcOffset: AllInputMarkers<X>` is not satisfied
+    /// // note: required by a bound in `FixedCalendarDateTimeFormatter::<C, FSet>::format`
+    /// formatter.format(&utc_offset);
+    /// ```
+    X,
```

Comment:
We should bikeshed the name of this.

Reply:
after merging sure

Reply:
did we ever bikeshed all the other names?

Reply:
I did an informal bikeshed with Manish one day...